packages/zoe/src/contracts/loan/index.js

```diff
@@ -63,24 +65,23 @@ const start = async zcf => {
     priceAuthority,
     periodNotifier,
     interestRate,
+    interestPeriod,
   } = zcf.getTerms();
 
   assert(autoswapInstance, `autoswapInstance must be provided`);
   assert(priceAuthority, `priceAuthority must be provided`);
   assert(periodNotifier, `periodNotifier must be provided`);
-  assert(interestRate, `interestRate must be provided`);
-
-  // TODO: make this non-lossy (notifier is lossy)
-  // https://github.com/Agoric/agoric-sdk/issues/2108
-  const periodAsyncIterable = makeAsyncIterableFromNotifier(periodNotifier);
+  assert(Nat(interestRate), `interestRate must be a positive integer`);
+  assert(Nat(interestPeriod), `interestPeriod must be a positive integer`);
```

Comment:
Since Nat throws, we won't be able to assert with the given message as it will never be reached:

```suggestion
// interestRate must be a positive integer
Nat(interestRate);
// interestPeriod must be a positive integer
Nat(interestPeriod);
```